proposals/0000-package-versioning.rst

```diff
@@ -0,0 +1,175 @@
+    approach to that. Any version bounds discussion must have the same
+    philosophy if it is to succeed: we need to acknowledge what authors have
+    been willing to do, and what we can reasonably educate people on
+
+    Authors make mistakes with the PVP (I had two package failures in the past
+    week because of it, even though I was blamed for one of them falsely).
+    Automated tooling will make mistakes with bounds. The PVP itself - even if
+    fully followed - does not guarantee 100% success. The only solution that
+    guarantees a 100% success of a build plan is curation (and even that is
+    brittle due to, e.g., different OSes). So we cannot reject a solution
+    because it won't work in some corner case. We need to minimize the
+    probability of failure, accepting that failure will inevitably happen.
+
+
+[1] https://gist.github.com/snoyberg/f6f10cdbea4b9e22d1b83e490ec59a10
+
+
+Motivation
+----------
+
+The problem addressed in this proposal is to reduce the burden on package
+maintainers who are currently required to update the implicit constraints when
+it becomes evident that subsequently published versions of a given dependency
+are indeed compatible with the current package.
+
+Failing to do so can cause the hackage solver to fail, resulting in not being
+able to use particular packages.
+
+Since the problemmatic constraints are essentially outside the control of the
+original developer, some way of moving the management of these out of the hands
+of the package maintainer is proposed.
+
+In Michael Snoyman's words
+
+* Let people continue adding in weird exceptions and cases that they know for
+  certain
+
+* Automate the thing most people are doing, and thereby ensure (1) even people
+  who don't care about the PVP are providing that info, (2) lower bound bumps
+  are not forgotten either, and (3) it's easier to test upper bound relaxes
+  going forward through automated tooling
+
+
+Proposed Change
+---------------
+
+The details of this section will be fleshed out as part of a discussion/dialogue
+with the interested parties.
+
+One possible solution is put forward
+`here <https://github.com/haskell/cabal/issues/3729>`_, but it probably
+requires too large a change to be practicable.
+
+Other solutions involve automatically setting the required implicit constraints
+on upload, using the `--pvp-bounds=both` option when uploading via stack, and a
+similar (to be added) feature for cabal.
+
+Or the implicit bounds could be calculated on hackage as part of the upload
+process, so that maintainers are then only required to specify constraints that
+have to be satisfied, failing which the package cannot be built.
+
+Other solutions ...
+
+The devil is in the details with each of these strategies. We should now being
+constructing concrete proposals around the alternatives, on the way to coming up
+with a single solution that works for all.
```

Comment:
While you are correct that no one wants buggy packages, as @alanz said, the problem is larger in scope. "Not building in the current snapshot" is a bug in stack/stackage, but *not* in general.